### PR TITLE
Add .gitignore into root project directory

### LINKED ISSUE #279: Consolidate `.gitignore` files for backend and frontend projects
**What would you like to be added**:
Currently, our project has separate `.gitignore` files for the backend and frontend projects. This makes it cumbersome to manage and maintain. It would be beneficial to consolidate the common entries into a global `.gitignore` file.

**Why is this needed**:
By consolidating the `.gitignore` entries into a global file, we can streamline the management of the project and ensure consistency in ignoring unnecessary files and directories across both the backend and frontend codebases. This will simplify the development process and reduce the chances of missing important entries.

**Additional Information**:

### PR DESCRIPTION
**Pull request by sihyeong671**
#### What this PR does / why we need it?
This PR adds a `.gitignore` file to the root directory. 

#### Any background context you want to provide?
The changes made in this PR are in relation to issue #279. The `.gitignore` file is crucial for specifying untracked files and directories to be ignored by Git.

#### What are the relevant tickets?

Fixes #279

### Checklist
- [x] Added relevant tests or not required
- [x] Didn't break anything

## Summary by CodeRabbit

- **New Features**
  - Introduced a comprehensive `.gitignore` file to improve project management by systematically ignoring unnecessary files across various development environments.
  - Enhanced coverage for Node.js, React, Visual Studio Code, and macOS, streamlining the development process.

### CODE REVIEW

**Comment by devleejb** [comment]
> [CLA assistant check](https://cla-assistant.io/yorkie-team/codepair?pullRequest=280) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/yorkie-team/codepair?pullRequest=280) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/yorkie-team/codepair?pullRequest=280) it.

Would you sign it?

**Comment by sihyeong671** [comment]
> > [CLA assistant check](https://cla-assistant.io/yorkie-team/codepair?pullRequest=280) Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/yorkie-team/codepair?pullRequest=280) before we can accept your contribution.You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/yorkie-team/codepair?pullRequest=280) it.
> 
> Would you sign it?

I didn't know I have to do this ... now I signed :)

**Review by devleejb — CHANGES_REQUESTED** [review]
It seems that many files are not related to CodePair. ex) next.js, nuxt, ...

And the comment of some directories such as `dist` are not for CodePair.
For example, 

```
# Nuxt.js build / generate output
dist
```

We use `dist` directory for built file in React and NestJS.

How about just combining two `.gitignore` in `frontend/` and `backend/`?

File: .gitignore [hunk 1]
```diff
@@ -1,42 +1,48 @@
...
 
-# IDEs and editors
-/.idea
+# Editor directories and files
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
+.DS_Store
```
Comment (devleejb) [inline .gitignore:21]: `.DS_Store` is for MacOS.
So the comment should be `# OS`.

File: .gitignore [hunk 2]
```diff
@@ -1,42 +1,48 @@
...
-# IDE - VSCode
-.vscode/*
-!.vscode/settings.json
-!.vscode/tasks.json
-!.vscode/launch.json
-!.vscode/extensions.json
 
 # Sqlite
 dev.db
 dev.db-journal
```
Comment (devleejb) [inline .gitignore:48]: This is legacy code.
Could you delete these three lines?

**Review by devleejb — CHANGES_REQUESTED** [review]

**Review by devleejb — APPROVED** [review]
Thank you for your contribution.

**Pull request merged**